charms/worker/k8s/src/charm.py

```diff
@@ -601,6 +601,44 @@ def _get_scrape_jobs(self):
             log.exception("Failed to get COS token.")
         return []
 
+    @on_error(
+        ops.WaitingStatus("Sharing Cluster Version"),
+        AssertionError,
+    )
+    def _update_kubernetes_version(self):
+        """Update the unit Kubernetes version in the cluster relation."""
+        if not (relation := self.model.get_relation("cluster")):
+            assert False, "Missing cluster integration"  # nosec
```

Comment:
Maybe this is a little clearer?  What's the `#nosec` covering here?
```suggestion
        relation = self.model.get_relation("cluster")
        assert relation "Missing cluster integration"  # nosec
```

Reply:
`nosec`: https://github.com/canonical/k8s-operator/pull/164#discussion_r1840866240